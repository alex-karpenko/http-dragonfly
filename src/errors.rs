use std::io;
use thiserror::Error;

#[derive(Error)]
pub enum HttpDragonflyError {
    #[error("unable to load config file '{}': {}", .filename, .cause)]
    LoadConfigFile { filename: String, cause: io::Error },
    #[error("unable to parse config: {}", .cause.kind)]
    ParseConfigFile {
        #[from]
        cause: figment::Error,
    },
    #[error("invalid config: {}", .cause)]
    ValidateConfig { cause: String },
}

impl std::fmt::Debug for HttpDragonflyError {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", self)
    }
}

#[cfg(test)]
mod tests {
    use std::io::ErrorKind;

    use figment::Error;
    use insta::assert_debug_snapshot;

    use super::*;

    #[test]
    fn errors() {
        assert_debug_snapshot!(HttpDragonflyError::LoadConfigFile {
            filename: "test-config.yaml".into(),
            cause: io::Error::new(ErrorKind::Other, "snapshot test cause")
        });
        assert_debug_snapshot!(HttpDragonflyError::ParseConfigFile {
            cause: Error::from("snapshot test cause".to_string())
        });
        assert_debug_snapshot!(HttpDragonflyError::ValidateConfig {
            cause: "snapshot test cause".to_string()
        });
    }
}
