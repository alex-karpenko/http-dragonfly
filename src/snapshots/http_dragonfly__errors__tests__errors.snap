---
source: src/errors.rs
expression: "HttpDragonflyError::LoadConfigFile {\n    filename: \"test-config.yaml\".into(),\n    cause: io::Error::new(ErrorKind::Other, \"snapshot test cause\"),\n}"
---
unable to load config file 'test-config.yaml': snapshot test cause
